compio-dispatcher/src/lib.rs

```diff
@@ -191,25 +190,23 @@ impl Dispatcher {
     /// thread panicked, this method will resume the panic.
     pub async fn join(self) -> io::Result<()> {
         drop(self.sender);
-        let results = Arc::new(Mutex::new(vec![]));
-        let event = Event::new();
-        let handle = event.handle();
+        let (tx, rx) = oneshot::channel::<Vec<_>>();
         if let Err(f) = self.pool.dispatch({
-            let results = results.clone();
             move || {
-                *results.lock().unwrap() = self
+                let results = self
                     .threads
                     .into_iter()
                     .map(|thread| thread.join())
                     .collect();
-                handle.notify();
+                tx.send(results).unwrap();
             }
```

Comment:
Do we really need `unwrap`? How about simply ignoring the result?

Reply:
You're right. Changed.